pkg/runner/step_context.go

```diff
@@ -478,13 +478,6 @@ func (sc *StepContext) runAction(actionDir string, actionPath string, localActio
 		}
 		actionName, containerActionDir := sc.getContainerActionPaths(step, actionLocation, rc)
 
-		sc.Env = mergeMaps(sc.Env, action.Runs.Env)
-
-		ee := sc.NewExpressionEvaluator()
-		for k, v := range sc.Env {
-			sc.Env[k] = ee.Interpolate(v)
-		}
-
```

Comment:
This action property is only supported for docker actions. I moved the code down to docker action code to fix the regression. Act should error out if it is used in other action types.